### PR TITLE
feat(question): add mark-as-useful button

### LINKED ISSUE #2959: [QuestionPage] Add "Mark as Useful" behaviour
Based on behaviour for Research/How-to content types.

### PR DESCRIPTION
**Pull request by goratt12**
PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Developer experience (improves developer workflows for contributing to the project)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Added "Mark as Useful" button on top of question, can be clicked only if logged-in

## Git Issues

Closes #2959

## Screenshots/Videos

image
image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Review by evakill — COMMENTED** [review]
cool, thanks @goratt12 - this is working well in preview and the code follows established patterns! couple small comments

File: src/models/question.models.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export namespace IQuestion {
     _createdBy: string
     _deleted: boolean
     subscribers?: UserIdList
+    votedUsefulBy?: UserIdList
```
[the remark this answers is not in the document]
Comment (goratt12) [reply src/models/question.models.tsx]: Correct, didn't notice that. I removed it :)

File: src/pages/Question/QuestionPage.tsx [hunk 2]
```diff
@@ -1,47 +1,75 @@
...
+  const onUsefulClick = async () => {
+    if (!store.activeUser?.userName) {
+      return null
+    }
+
+    // Trigger update without waiting
+    const questionId = store.activeQuestionItem?._id
+    if (questionId) {
+      store.toggleUsefulByUser(questionId, store.activeUser?.userName)
+      store.userVotedActiveQuestionUseful
```
[the remark this answers is not in the document]
Comment (goratt12) [reply src/pages/Question/QuestionPage.tsx:41]: What line?
`setQuestion(store.activeQuestionItem)`
without it the component doesn't re-render.
what is the recommended way of doing that?

[the remark this answers is not in the document]
Comment (goratt12) [reply src/pages/Question/QuestionPage.tsx]: No, it does nothing at all 😂
I removed it

**Review by davehakkens — APPROVED** [review]

**Pull request merged**